internal/server/auth/middleware.go

```diff
@@ -176,6 +177,52 @@ func EmailMatchingInterceptor(logger *zap.Logger, rgxs []*regexp.Regexp) grpc.Un
 	}
 }
 
+func NamespaceMatchingInterceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		auth := GetAuthenticationFrom(ctx)
+
+		if auth == nil {
```

Comment:
I wonder if we should panic here instead?
My thought being we should fail fast if this has accidentally been installed in the wrong order.

Reply:
We could thread in the same interceptor options above and do the explicit skipped checked for the skipped services.